src/game/level.cpp

```diff
@@ -10,6 +10,185 @@ void Level::load_json(std::vector<std::shared_ptr<Entity>> *entities, const std:
+  std::string layer_id;
+
+  for (u32 i = 0; i < (json_data_size - 14); i++) {
+    if (i == 0)
+      layer_id = "/layers";
+    else
+      layer_id = "/layers" + std::to_string(i);
+
+    simdjson::ondemand::array layers = this->json_data.at_pointer(layer_id).get_array();
+
+    simdjson::ondemand::array background_array = layers.at_pointer("/0/chunks").get_array();
+
+    u32 bg_array_size = background_array.count_elements();
+
+    for (u32 i = 0; i < bg_array_size; i++) {
+      std::string json_pointer = layer_id + "/0/chunks/" + std::to_string(i);
+
+      simdjson::ondemand::object obj(this->json_data.at_pointer(json_pointer).get_object());
+
+      f64 x = obj["x"];
+      f64 y = obj["y"];
+
+      std::vector<u32> values;
+
+      simdjson::ondemand::array arr((obj["data"].get_array()));
+
+      for (f64 value : arr)
+        values.push_back(value);
+
+      this->background_content.emplace_back(x, y, std::move(values));
+    }
+
+    simdjson::ondemand::array objects_array = this->json_data.at_pointer(layer_id + "/1/objects").get_array();
+    u32 obj_array_size                      = objects_array.count_elements();
+
+    for (u32 i = 0; i < obj_array_size; i++) {
+      std::string json_pointer = layer_id + "/1/objects/" + std::to_string(i);
+
+      simdjson::ondemand::object obj(this->json_data.at_pointer(json_pointer).get_object());
+
+      f64 x          = obj["x"];
+      f64 y          = obj["y"];
+      f64 width      = obj["width"];
+      f64 height     = obj["height"];
+      f64 tileset_id = obj["id"];
+
+      simdjson::ondemand::array property_array = this->json_data.at_pointer(json_pointer + "/properties").get_array();
+      u32 prop_array_size                      = property_array.count_elements();
+      std::string script                       = "return";
+
+      for (u32 i = 0; i < prop_array_size; i++) {
+        simdjson::ondemand::object obj(property_array.at_pointer("/" + std::to_string(i)).get_object());
+
+        script = std::string_view(obj["value"]);
+      }
+
+      objects_content.emplace_back(x, y, width, height, tileset_id, script);
+    }
+
+    simdjson::ondemand::array collisions_array = this->json_data.at_pointer(layer_id + "/2/chunks").get_array();
+
+    u32 co_array_size = collisions_array.count_elements();
+
+    for (u32 i = 0; i < co_array_size; i++) {
+      std::string json_pointer = layer_id + "/2/chunks/" + std::to_string(i);
+
+      simdjson::ondemand::object obj(this->json_data.at_pointer(json_pointer).get_object());
+
+      i64 x = obj["x"];
+      i64 y = obj["y"];
+
+      std::vector<u64> values;
+
+      simdjson::ondemand::array arr((obj["data"].get_array()));
+
+      for (u64 value : arr)
+        values.push_back(value);
+
+      collisions_content.emplace_back(x, y, std::move(values));
+    }
+  }
+
+  simdjson::ondemand::array tilesets = this->json_data["tilesets"];
+  u32 tilesets_size                  = tilesets.count_elements();
```

Comment:
you don't need this variable, just use the function directly:
`for (u32 i = 14; i < tilesets.count_elements(); i++) {`